### ISSUE TITLE
crash when changing size of toolbar icons

### ISSUE THREAD

**Issue by Dobby233Liu** [d2lmirrors/anolis#36]
*creation date = `2009-03-29T20:23:57.883-07:00`*

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.
 
************** Exception Text **************
System.UnauthorizedAccessException: Access to the registry key 'HKEY_CLASSES_ROOT\exefile' is denied.
   at Microsoft.Win32.RegistryKey.Win32Error(Int32 errorCode, String str)
   at Microsoft.Win32.RegistryKey.CreateSubKey(String subkey, RegistryKeyPermissionCheck permissionCheck, RegistrySecurity registrySecurity)
   at Anolis.Resourcer.Settings.FileAssociationManager.AssociateWithFiles(String[] classes, Boolean isAssociated) in D:\Users\David\My Documents\Visual Studio Projects\Anolis\Anolis.Resourcer\Settings\FileAssociationManager.cs:line 71
   at Anolis.Resourcer.Settings.Settings.AssociateWithFiles(Boolean isAssociated) in D:\Users\David\My Documents\Visual Studio Projects\Anolis\Anolis.Resourcer\Settings\Settings.cs:line 68
   at Anolis.Resourcer.OptionsForm.__ok_Click(Object sender, EventArgs e) in D:\Users\David\My Documents\Visual Studio Projects\Anolis\Anolis.Resourcer\Forms\OptionsForm.cs:line 80
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
 
 
************** Loaded Assemblies **************
mscorlib
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.50727.4908 (NetFXspW7.050727-4900)
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v2.0.50727/mscorlib.dll
----------------------------------------
Anolis.Resourcer.Merged
    Assembly Version: 0.8.3363.34208
    Win32 Version: 0.8.2.0
    CodeBase: file:///C:/Users/Digerati/Desktop/Anolis.Resourcer.Merged.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.50727.4908 (NetFXspW7.050727-4900)
    CodeBase: file:///C:/Windows/assembly/GAC_MSIL/System.Windows.Forms/2.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.50727.4908 (NetFXspW7.050727-4900)
    CodeBase: file:///C:/Windows/assembly/GAC_MSIL/System/2.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.50727.4908 (NetFXspW7.050727-4900)
    CodeBase: file:///C:/Windows/assembly/GAC_MSIL/System.Drawing/2.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.50727.4908 (NetFXspW7.050727-4900)
    CodeBase: file:///C:/Windows/assembly/GAC_MSIL/System.Configuration/2.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.50727.4908 (NetFXspW7.050727-4900)
    CodeBase: file:///C:/Windows/assembly/GAC_MSIL/System.Xml/2.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
 
************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.
 
For example:
 
<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>
 
When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

**Comment by Dobby233Liu** [comment]
*creation date: `2009-03-30T06:47:03.64-07:00`*

This crash isn't caused by changing the toolbar size, but actually when setting the file association.

I forgot that on Vista you need administrative privileges to do this. I'll fix this in the next release.

**Comment by Dobby233Liu** [comment]
*closed date = `2009-04-10T12:42:30.48-07:00`*

This is now fixed. It'll be in the next release of Resourcer,
